lib/src/main/java/com/diffplug/spotless/json/JsonFormatterStep.java

```diff
@@ -0,0 +1,111 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.json;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Objects;
+
+import com.diffplug.spotless.FormatterFunc;
+import com.diffplug.spotless.FormatterStep;
+import com.diffplug.spotless.JarState;
+import com.diffplug.spotless.Provisioner;
+
+public final class JsonFormatterStep {
+	private static final String MAVEN_COORDINATE = "org.json:json:";
+	private static final String DEFAULT_VERSION = "20210307";
+
+	public static FormatterStep create(Integer indent, Provisioner provisioner) {
+		Objects.requireNonNull(provisioner, "provisioner cannot be null");
+		return FormatterStep.createLazy("json", () -> new State(indent, provisioner), State::toFormatter);
+	}
+
+	private static final class State implements Serializable {
+		private static final long serialVersionUID = 1L;
+
+		private final int indentSpaces;
+		private final boolean useTabs;
+		private final JarState jarState;
+
+		private State(Integer indent, Provisioner provisioner) throws IOException {
+			if (indent != null) {
+				this.indentSpaces = indent;
+				this.useTabs = false;
+			} else {
+				this.indentSpaces = 20; // it's unlikely that someone will use this many spaces, so should be safe
+				useTabs = true;
+			}
+
+			this.jarState = JarState.from(MAVEN_COORDINATE + DEFAULT_VERSION, provisioner);
+		}
+
+		FormatterFunc toFormatter() {
+			Method objectToString;
+			Method arrayToString;
+			Constructor<?> objectConstructor;
+			Constructor<?> arrayConstructor;
+			try {
+				ClassLoader classLoader = jarState.getClassLoader();
+				Class<?> jsonObject = classLoader.loadClass("org.json.JSONObject");
+				Class<?>[] constructorArguments = new Class[]{String.class};
+				objectConstructor = jsonObject.getConstructor(constructorArguments);
+				objectToString = jsonObject.getMethod("toString", int.class);
+
+				Class<?> jsonArray = classLoader.loadClass("org.json.JSONArray");
+				arrayConstructor = jsonArray.getConstructor(constructorArguments);
+				arrayToString = jsonArray.getMethod("toString", int.class);
+			} catch (ClassNotFoundException | NoSuchMethodException e) {
+				throw new IllegalStateException("There was a problem preparing org.json dependencies", e);
+			}
+
+			return s -> {
+				String prettyPrinted = null;
+				if (s.isEmpty()) {
+					prettyPrinted = s;
+				}
+				if (s.startsWith("{")) {
+					try {
+						Object parsed = objectConstructor.newInstance(s);
+						prettyPrinted = objectToString.invoke(parsed, indentSpaces) + "\n";
+					} catch (InvocationTargetException ignored) {
+						// ignore if we cannot convert to JSON string
```

Comment:
Swallowing exceptions always leads to pain. Remove the try/catch.

Reply:
Although it's swallowed, we still throw an `AssertionError` to show there's nothing formatted - would you prefer it throws the original `InvocationTargetException`?